pkg/compact/compact.go

```diff
@@ -958,16 +958,18 @@ func (c *BucketCompactor) Compact(ctx context.Context) (rerr error) {
 		}
 
 		level.Info(c.logger).Log("msg", "start of compactions")
+		st := time.Now()
+		compactCount := 0
 
 		// Send all groups found during this pass to the compaction workers.
 		var groupErrs terrors.MultiError
-	groupLoop:
 		for _, g := range groups {
 			select {
 			case groupErr := <-errChan:
 				groupErrs.Add(groupErr)
-				break groupLoop
```

Comment:
I kinda agree on not breaking, but I think we should continue breaking in case of `IsHaltError(groupErr)`.

Reply:
Thanks for your comment, en, I think it is a good way to continue compact
- one loop, compact all group as expected, for lot's of block, this is very useful(if break on any halt error, the following groups may not be executed at once)
- halt or retry will decide in external function (whether alarm, etc.), see the screenshot below

![image](https://user-images.githubusercontent.com/1853121/85188487-86f3e680-b2d9-11ea-8228-f02a2688b4d7.png)